simpleviewinc/tap-events-force
# Zen 563 filter modal

**Ticket**: https://jira.simpleviewtools.com/browse/ZEN-563

## Context

When certain labels do not exist , the option to filter by that label should not appear. Also , when there is nothing that can actually be filtered , the Filter button is unnecessary

## Goal

* Ensure that when for all sessions 'isWaitingListAvailable' option is set to false , the 'Waiting List' and 'On waiting List' options do not appear in the filter labels
* Ensure that when for all sessions 'allowBooking' option is set to false , the 'Available' and 'Selected' options do not appear in the filter labels
* Ensure that when there are no custom labels AND there are no sessions with a true 'isWaitingListAvailable' or 'allowBooking' option , the filter option/button is not present since there is nothing to filter by

## Updates

* First part of updates was code to ensure 'Available' and 'Selected' appeared / disappeared based on the incoming data allowBooking and the second part was to ensure filter option disappeared when there were no labels
* *tap-events-force/src/hooks/sessions/useAllowBookingActive.js*
  * Hook to update the store object based on AllowBookingActive
* *tap-events-force/src/components/modals/filter.js*
  * Apply logic appropriately to show 'Available' and 'Selected' OR 'Waiting List' and 'On waiting List'  OR both OR none based on AllowBookingActive and WaitingListActive
* *tap-events-force/src/actions/session/filters/applySessionFilters.js*
  * Changes to apply session filters by also calling AllowBookingActive
* *tap-events-force/src/actions/session/booking/setAllowBookingActive.js*
  * Test the boolean value of the allowBooking field at the root of the session data object 
* *tap-events-force/src/components/sessionsHeader/sessionsHeader.js*
  * Changes to display the 'Right Component' which is the parent of the Filter button , only when some labels to filter on are present

## Testing
*Setup
- Run the command => keg evf pack run evf:zen-563-filtermodal
- Go to http://evf-ZEN-563-FilterModal.local.keghub.io

Note 1: Docker image : docker pull ghcr.io/simpleviewinc/evf:zen-563-filtermodal
Note 2: Images for testing attached at the end

* Test 1 : To confirm if Waiting List functionality is working
  * Launch events force from the docker image above
  * Click on Filter link on top right and confirm that 'Waiting List' and 'On Waiting List' labels are present
  * Click on the 'Test Data (JSON)' button at the top left
  * Search 'Sessions' section , and within that find every session where isWaitingListAvailable: true and set it to false, Apply changes
  * Click the Filter link again and confirm that 'Waiting List' and 'On Waiting List'  are no longer present

* Test 2 : To confirm if Select/Available functionality is working
  * Launch events force from the docker image above
  * Click on Filter link on top right and confirm that 'Available' and 'Selected' labels are present
  * Click on the 'Test Data (JSON)' button at the top left
  * Search 'Sessions' section , and within that find every session where allowBooking: true and set it to false, Apply changes
  * Click the Filter link again and confirm that 'Available' and 'Selected' are no longer present

* Test 3 : To confirm if Filter link disappear functionality is working
  * Launch events force from the docker image above
  * Click on Filter link on top right and confirm that all labels are present. Custom labels at the top section and default lables at the bottom section of the modal
  * Click on the 'Test Data (JSON)' button at the top left
  * Search 'Sessions' section , and within that find every session where allowBooking: true and set it to false, Apply changes
  * Search 'Sessions' section , and within that find every session where isWaitingListAvailable: true and set it to false, Apply changes
  * Find the labels section and comment it out entirely
  * These 3 steps means there are no labels that can be valid filters. Therefore there is no necessity for the filter link
  * Ensure Filter link is no longer present

Attached below are images taken from my testing and that will assist in yours
*Image 1: Starting with all labels / filter options present
![ScreenShot-1](https://user-images.githubusercontent.com/80443383/128076598-4f73ac37-ae10-4658-9a60-a5cbcb26d9fe.png)

*Image 2: When there are no isWaitingListAvailable: true  but atleast one allowBooking: true
![ScreenShot-2](https://user-images.githubusercontent.com/80443383/128076615-76f6562f-fb0c-4682-a8c8-6e83acc73f16.png)

*Image 3: When there are no allowBooking: true but atleast one isWaitingListAvailable: true 
![ScreenShot-3](https://user-images.githubusercontent.com/80443383/128076632-59459a91-d423-4e6a-b71d-4575446d7962.png)

*Image 4: When there are no isWaitingListAvailable: true OR allowBooking: true
![ScreenShot-4](https://user-images.githubusercontent.com/80443383/128076640-74b6a26a-bee2-49c7-a94e-f1b3e19a4a10.png)

*Image 5: When there are no custom labels but default labels are present because of atleast one allowBooking: true and isWaitingListAvailable: true
![ScreenShot-5](https://user-images.githubusercontent.com/80443383/128076667-bac5bf1c-09d1-45d5-a483-273ddf63faf9.png)

*Image 6: When there are no custom labels OR default labels present 
![ScreenShot-6](https://user-images.githubusercontent.com/80443383/128076690-c382523c-f020-4edd-8a4e-eb687f3cec9e.png)
